specs/daito/OpenAPI_Design.md

```diff
@@ -0,0 +1,164 @@
+# VMware Ecosystem OpenAPI
+
+**Author(s)**: [zengyingzhe](https://github.com/zengyingzhe)
+
+## Summary
+
+This documentation serves as the design spec for VMware OpenAPI.
+VMware OpenAPI is a framework which provides generic storage related APIs to upper VMware plugins.
+
+## Motivation
+
+VMware is one of the most important and popular virtualization platforms that depends on storage, so it is very necessary to integrate OpenSDS to VMware platform, and this will also benefit for OpenSDS' further development.
+To achieve various purposes, VMware supports multiple types of plugins. All of them rely on storage APIs to interact with backend storage system.
+So if we implement a common framework that encapsulates API details of backend storage and provisions more generic interfaces, it would be much easier for plugin developers to implement a plugin, because developers can concern more to plugin itself's logic, not stuck in communication with storage.
+
+### Goals
+
+* Provide northbound generic APIs that encapsulate storage northbound APIs(through REST or others).
+* Open scalable&pluggable framework, 3rd parties can encapsulate its own storage's APIs following northbound interfaces' definition.
+
+### Non-Goals
+
+None.
+
+## Design Details
+
+### OpenAPI Usage Architecture
+
+![](media/OpenAPI_Usage_Architecture.png)
+
+VMware OpenAPI is a language dependant library, such as a Jar package for Java.
+VMware plugins should import the OpenAPI library and call its northbound interfaces to communicate with corresponding storage.
```

Comment:
Bit confusing here! are we devolping both VMware Plugins and openAPI library? Can you please eloborate what is Vmware plugin and openAPI and scope of both?

Reply:
We're only develping openapi here.
Thanks for pointing out.
Already add some explaination.